status-im/status-react
# Push notifications: iOS request permissions, surface token and flags

### Summary
Push notifications: iOS request permissions, surface token and flags

- Asks for permissions on iOS at startup
- Introduce NOTIFICATIONS_WIP_ENABLED flag
- Surface FCM Token and put it in clipboard for QA manual end to end testing

### Note to reviewers

This is a WIP and part of spiking out push notifications end to end. The goal of this is to let QA do end to end testing on devices and surface problems now rather than later. The next step is to integrate with status-go code to trigger notifications on messages from that side, as it appears more desirable than doing it from RN side (via RN Firebase Cloud Functions). This will be addressed in future issues and PRs.

### Steps to test for QA
1. Load app and login, you should see a popup with a token. If on iOS it'll ask you for permissions, grant it.

2. The token is in your clipboard, paste it so it ends up at your computer.

3. At your computer, go to https://console.firebase.google.com/u/1/ under Status account, should see Status React App. If you have issues with this please message me.

4. Background the app.

5. Grow -> Notifications -> New message.

6. Enter message text, target single device and paste in FCM registration token and send message.

7. You should see a notification and message completed at Firebase.

Note: iOS works fine, Android seems to depend on Google Play Service. I suspect there are some issues there so logs would be appreciated. I don't think we'll be able to get it to work for all Android devices/setups with just Firebase, though. So consider this a work in progress.

status: ready
